### ISSUE TITLE
Pass-through income MTR outliers

### ISSUE THREAD

**Issue by rickecon** [open-source-economics/Tax-Calculator#1636]
The Tax Cuts and Jobs Act has some changes to pass through income that cause some huge outlier negative marginal tax rates in self employment income (`e00900p`). These influence the marginal tax rates of self employment income. When the pass through variables in the [`TCJA_Nov3.json`](?) are passed into Tax-Calculator `[mtr_fica_sey, mtr_iit_sey, mtr_combined_sey] = calc1.mtr('e00900p')`. When I describe the variable `mtr_combined_sey`, which I will call `MTR_sey` ("MTR self employment income"), I get the following.
```python
[In] MTR_sey[MTR_sey >= -0.45].describe()
[Out]
count         212111
mean          0.278268
std           0.140092
min          -0.416316
25%           0.250000
50%           0.252818
75%           0.373634
max           1.054188

[In] MTR_sey[MTR_sey < -0.45].describe()
[Out]
count    3414
mean    -5.900011e+06
std      1.152289e+07
min     -2.274409e+08
25%     -6.169098e+06
50%     -2.822819e+06
75%     -9.383364e+05
max     -4.500000e-01
```
The average MTR for all the values with MTR < -0.45 is -5.9e+06 or -59,000 percent. I'm sure these are coming from the marginal taxpayers at some big ledges, but we might want to do some smoothing here. @martinholmer @MattHJensen @jdebacker

**Comment by martinholmer** [comment]
@rickecon [reported](https://github.com/open-source-economics/Tax-Calculator/issues/1636#issue-272491029) 3414 observations in `puf.csv` that have a MTR wrt e00900 that is less than -45% under the `TCJA_Nov3.json` reform.  Then he said:
> The average MTR for all the values with MTR < -0.45 is -5.9e+06 or -59,000 percent. I'm sure these are coming from the marginal taxpayers at some big ledges, but we might want to do some smoothing here. 

It seems to me that we need answers to a few questions (assuming you're using taxcalc 0.13.0):
- What does this MTR distribution look like under current-law policy?
- Can you provide us with the RECID for one or two cases that have an extreme negative MTR?
- Can you, or anybody else, explain what "big ledges" are in the `TCJA_Nov3.json` reform?

@MattHJensen @jdebacker @codykallen

**Comment by martinholmer** [comment]
As a background note, the distribution of the MTR wrt `e00900p` under current-law policy has not changed since June 27, 2017.  Here is that MTR distribution (from the `taxcalc/tests/pufcsv_mtr_expect.txt` file):
```
PTAX mtr histogram bin edges:
     [0.0, 0.02, 0.04, 0.06, 0.08, 0.1, 0.12, 0.14, 0.16, 0.18, 1.0]
ITAX mtr histogram bin edges:
     [-1.0, -0.3, -0.2, -0.1, 0.0, 0.1, 0.2, 0.3, 0.4, 0.5, 1.0]
PTAX and ITAX mtr histogram bin counts for e00200p:
215525 :      0  30324      0      0      0      0      0 180117   5084      0
215525 :   5003    105   1747  11294  34520  66930  45037  28123  22114    652
PTAX and ITAX mtr histogram bin counts for e00200s:
108115 :      0  14748      0      0      0      0      0  83207  10160      0
108115 :    979     33    651   3161  12126  28352  23899  20478  17988    448
PTAX and ITAX mtr histogram bin counts for e00900p:
215525 :  20799  25348      0      0      0      0      0 169378      0      0
215525 :   5003    105   1746  11158  47533  56438  51408  24431  17388    315
```
Notice that there are 5003 records with a MTR wrt e00900p between -100% and -30%.

And also notice that the sum of `5003 105 ... 315` is 215525, which I believe is the `puf.csv` count, which means there are no records with a MTR below -100%.

Looking back at the history of this file, there have been 3000 to 5000 records with MTR wrt e00900p for the past two years.  Nobody has complained about those values, probably because it looks as if these negative MTR wrt e00900p values are the same as MTR wrt e00200p (earnings), which suggests they are generated through the workings of the EITC.

But now under the `TCJA_Nov3.json` reform there are some records with MTR below -100% according to the report from @rickecon.  Does that seem sensible?   I guess there are two possibilities:
- The reform really does create a tax system in which an **increase** in e00900p can **reduce** income taxes by more than the increase
- We don't have the logic of the reform implemented correctly in taxcalc 0.13.0

Do people have any thoughts about which option is true?

@MattHJensen @feenberg @rickecon @jdebacker @Amy-Xu @andersonfrailey @hdoupe @codykallen 
@GoFroggyRun

**Comment by codykallen** [comment]
Upon inspection, I have identified the reason that this occurs: `_PT_wages_active_income`.  This policy parameter applies only if an individual has active pass-through income (`e00900` or `e26270`). When the marginal tax rate on `e00900p` is calculated and `_PT_wages_active_income == True`, then individuals who do not have active Sch C income in the PUF suddenly have their wages eligible for the pass-through rate. For individuals subject who have wage income subject to the 35% rate or higher, this would substantially decrease their taxes owed. 

I think the solution to this should be to add an additional line of code, such that the amount of active income to which `_PT_EligibleRate_active` is applied may not exceed `e00900 + e26270`. This should eliminate the problem. 

If others believe this is an appropriate solution, I can implement it fairly quickly.

**Comment by codykallen** [comment]
Correction: In my previous comment, I suggested:

> I think the solution to this should be to add an additional line of code, such that the amount of active income to which `_PT_EligibleRate_active` is applied may not exceed `e00900 + e26270`.

That should say:

> I think the solution to this should be to add an additional line of code, such that the amount of active income eligible for the pass-through rates (after `_PT_EligibleRate_active` is applied) may not exceed `e00900 + e26270`.

**Comment by martinholmer** [comment]
@codykallen said about MTR wrt e00900p less than -100% under the TCJA.json reform:
> I think the solution to this should be to add an additional line of code, such that the amount of active income eligible for the pass-through rates (after _PT_EligibleRate_active is applied) may not exceed e00900 + e26270.

Here's how I suggest you proceed:
- Add a new test (using the puf_subsample) that fails because it generates at least one MTR(e00900p) less than minus 100% under the TCJA.json reform
- Add your extra line of code
- Rerun the test to make sure that your extra line of code allows the new test to pass

Does this make sense?

**Comment by codykallen** [comment]
@martinholmer, PR #1645 should resolve this issue. It also adds a new test, which fails on the master branch but passes with the PR's branch.

**Comment by martinholmer** [comment]
This issue has been resolved by merged pull request #1645.
